data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/SourcePartitionStoreItem.java

```diff
@@ -20,4 +20,12 @@ public interface SourcePartitionStoreItem {
     Instant getPartitionOwnershipTimeout();
```

Comment:
How do the `SourcePartitionStoreItem` and `SourcePartition` differ in usage? At a glance it seems like they are both representations of data in the partition store, I'm wondering if there's room to reconcile them into a single object